Wireshark/plugins/v2gtlssecret.lua

```diff
@@ -115,27 +132,40 @@ function p_v2gtlssecret.dissector(buf,pinfo,root)
         -- check if the TLS secrets are already in the file
         local file, _, _ = io.open(get_preference("tls.keylog_file"), "r")
         if file ~= nil then
-            for line in file:lines() do
-                local tls_secret_of_file = tostring(line)
-                for idx = #tls_secret_list, 1, -1 do
-                    if tls_secret_list[idx] == tls_secret_of_file then
-                        table.remove(tls_secret_list, idx)
+            local file_content = file:read("*a")
+            file:close(file)
+    
+            for idx = #tls_secret_list, 1, -1 do
+                local to_be_removed = false
+                local splitted_from_packet = split_string(tls_secret_list[idx])
+                for line in file_content:gmatch'[^\r\n]+' do    
+                    local splitted_from_file = split_string(tostring(line))
+                    if #splitted_from_packet == 3 and #splitted_from_file == 3 then
+                        if splitted_from_file[1] == splitted_from_packet[1] and splitted_from_file[2] == splitted_from_packet[2] then
+                            if splitted_from_file[3] == splitted_from_packet[3] then
+                                to_be_removed = true
+                            else
+                                add_expert_info("CLIENT RANDOM is not unique!", subtree, pinfo, ef_io_error)
```

Comment:
Maybe `splitted_from_packet[1]` instead of `CLIENT RANDOM`?

Reply:
It is actually the CLIENT RANDOM part of the secret that is not unique here, even if the label is e.g. SERVER_TRAFFIC_SECRET. However, i'll improve the warning.